**An Introduction to Statistical Learning with Applications in R**

**Punto 3 Taller - página 124 (punto 10)**

William Martinez

William Gomez

Laura Gonzalez

In [2]:
install.packages('ISLR2')
library(ISLR2)

Data<- Carseats
head(Data,2)

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



Sales CompPrice Income Advertising Population Price ShelveLoc Age Education
1  9.50 138       73     11          276        120   Bad       42  17       
2 11.22 111       48     16          260         83   Good      65  10       
  Urban US 
1 Yes   Yes
2 Yes   Yes

**(a) Ajusta un modelo de regresión múltiple para predecir las Ventas usando Precio, Urbano y EE.UU.**



In [ ]:
# Ajustar el modelo de regresión múltiple
modelo <- lm(Sales ~ Price + Urban + US, data = Carseats)
summary(modelo)


Call:
lm(formula = Sales ~ Price + Urban + US, data = Carseats)

Residuals:
    Min      1Q  Median      3Q     Max 
-6.9206 -1.6220 -0.0564  1.5786  7.0581 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) 13.043469   0.651012  20.036  < 2e-16 ***
Price       -0.054459   0.005242 -10.389  < 2e-16 ***
UrbanYes    -0.021916   0.271650  -0.081    0.936    
USYes        1.200573   0.259042   4.635 4.86e-06 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 2.472 on 396 degrees of freedom
Multiple R-squared:  0.2393,	Adjusted R-squared:  0.2335 
F-statistic: 41.52 on 3 and 396 DF,  p-value: < 2.2e-16


**(b) Proporciona una interpretación de cada coeficiente en el modelo. ¡Ten cuidado, algunas de las variables en el modelo son cualitativas!**



Un modelo de regresión múltiple se ha ajustado para predecir las ventas (**Sales**) usando las variables **Price**, **Urban**, y **US**. Se han obtenido los siguientes resultados estadísticos relevantes de dicho ajuste:

El intercepto, Price y USYes tienen un nivel de **significancia del 0.1%**, lo que indica que estas variables y el intercepto tienen una relación estadísticamente significativa con la variable dependiente (Sales). Es decir, estas variables muestran un comportamiento lineal con respecto a las ventas.

Estos resultados sugieren que:

- El intercepto es de 13.04.

- La variable Price tiene una relación inversa con las ventas (Sales), lo que significa que a medida que el precio aumenta, las ventas tienden a disminuir.

- La variable USYes tiene una relación positiva con las ventas, lo que indica que las ventas son más altas en Estados Unidos en comparación con otros países.

En este modelo, las variables categóricas Urban y US se han convertido automáticamente en variables **dummies**, que al ser binarias, permiten comparar entre dos categorías. Solo una de las dos categorías es necesaria para representar la variable en el modelo, ya que cuando toma el valor de 1 representa "Yes", y cuando toma el valor de 0 representa "No".

Por otro lado, los **residuos**, que representan las diferencias entre los valores observados de Sales y los valores predichos por el modelo, tienen una mediana cercana a 0, lo cual es deseable, ya que indica que el error promedio del modelo es bajo.

El problema de este modelo es el **R² ajustado**, que es de solo 0.2335, lo que significa que nuestro modelo, con las variables seleccionadas, es capaz de explicar únicamente el 23.35% de la variabilidad en las ventas (Sales).

**(c) Escribe el modelo en forma de ecuación, manejando adecuadamente las variables cualitativas.**



Ecuación del Modelo
La ecuación del modelo de regresión múltiple, considerando las variables cualitativas, es:

**Sales = 13.043469 − (0.054459×Price) − (0.021916×UrbanYes) + (1.200573×USYes)**


Interpretación:
Si Urban = No y US = No (las categorías de referencia),dichas variables toman el valor de 0 y la ecuación se reduce a:

**Sales = 13.043469 − (0.054459×Price)**

Si Urban = Yes, entonces dichas variables toman el valor de 1 y la ecuacion se convierte en:

Sales = 13.043469 − (0.054459×Price) - (0.021916 * 1) + (1.200573 * 1)

**Sales = 14.222126 − (0.054459×Price)**

- Intercepto: Ventas promedio cuando Price = 0, Urban = No, y US = No.

- Price: Impacto negativo del precio en las ventas.

- UrbanYes: No hay un impacto significativo de estar en una área urbana en comparación con una no urbana.

- USYes: Impacto positivo significativo de estar en EE.UU. en comparación con estar en otro país.

**(d) ¿Para cuál de los predictores puedes rechazar la hipótesis nula
𝐻0:𝛽𝑗=0?**




H0:βj=0 para todos los predictores j

H1:βj=0 para al menos un predictor j

La prueba de hipótesis en regresión lineal determina si cada predictor tiene un impacto significativo en la variable dependiente. Si el valor p es menor que el nivel de significancia, se concluye que el predictor afecta significativamente la variable dependiente, y se rechaza H0.

Para determinar si podemos rechazar la hipótesis nula para cada predictor, observamos el valor p (p-value) asociado a cada coeficiente en el modelo:

- Intercepto:p<2e-16 (muy significativo)

- Price:p<2e-16 (muy significativo)

- UrbanYes:p=0.936 (no significativo)

- USYes:p=4.86e-06 (muy significativo)

**Conclusión**: Podemos rechazar la hipótesis nula
H
0
​
 :β
j
​
 =0 para los predictores Price y USYes, ya que sus valores p son muy bajos (menos de 0.05). Esto quiere decir que hay evidencia suficiente para sugerir que Price y USYes

**(e) Basándote en tu respuesta a la pregunta anterior, ajusta un modelo más pequeño que solo utilice los predictores para los cuales hay evidencia de asociación con el resultado.**



In [ ]:
modelo_reducido <- lm(Sales ~ Price + US, data = Carseats)
summary(modelo_reducido)


Call:
lm(formula = Sales ~ Price + US, data = Carseats)

Residuals:
    Min      1Q  Median      3Q     Max 
-6.9269 -1.6286 -0.0574  1.5766  7.0515 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) 13.03079    0.63098  20.652  < 2e-16 ***
Price       -0.05448    0.00523 -10.416  < 2e-16 ***
USYes        1.19964    0.25846   4.641 4.71e-06 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 2.469 on 397 degrees of freedom
Multiple R-squared:  0.2393,	Adjusted R-squared:  0.2354 
F-statistic: 62.43 on 2 and 397 DF,  p-value: < 2.2e-16


**(f) ¿Qué tan bien se ajustan los modelos en (a) y (e) a los datos?**


En términos de ajuste, podemos comparar los modelos en (a) y (e) con base en las métricas de ajuste, como el R-cuadrado ajustado y el error estándar residual.


---

**Modelo (a): Sales ~ Price + Urban + US**

**R-cuadrado ajustado:** 0.2335

**Error estándar residual:** 2.472

---


**Modelo (e): Sales ~ Price + US**

**R-cuadrado ajustado:** 0.2354

**Error estándar residual:** 2.469

---


Comparación de los modelos
**R-cuadrado ajustado:** El modelo (e) tiene un R-cuadrado ajustado de 0.2354, mientras que el modelo (a) tiene un R-cuadrado ajustado de 0.2335. Aunque el modelo (e) explica una fracción marginalmente mayor de la variabilidad en Sales, la diferencia es muy pequeña, lo que sugiere que ambos modelos tienen un poder explicativo similar y ninguno de los dos explica significativamente más variabilidad en Sales.

**Error estándar residual**: El modelo (e) tiene un error estándar residual ligeramente menor (2.469 vs. 2.472), lo que sugiere un ajuste un poco más preciso de los datos.


---


**CONCLUSIÓN**
El modelo reducido (e) que solo incluye Price y US ajusta los datos tan bien como el modelo completo (a), e incluso mejor en términos de simplicidad y significancia estadística de los coeficientes. Al excluir Urban, se logra un modelo más parsimonioso sin perder capacidad predictiva significativa.

**FIN**